### PR TITLE
Upgrade to pandoc 2.0

### PR DESCRIPTION
**Pull request by dhimmel**
Release notes at:
https://github.com/jgm/pandoc/releases/tag/2.0
http://pandoc.org/releases.html#pandoc-2.0-29-oct-2017

### CODE REVIEW

**Comment by dhimmel** [comment]
Note that there could be problems with `pandoc-2.0-1-amd64.deb` (see https://github.com/jgm/pandoc/issues/4006). We should wait for that issue to be resolved before merging.

**Comment by janschulz** [comment]
Sounds good, lets wait for the next release. Rest is lgtm!

**Comment by dhimmel** [comment]
Pandoc replaced the linux build with `pandoc-2.0-2-amd64.deb` to address the issue above. I changed the templated URL for linux to:

```
https://github.com/jgm/pandoc/releases/download/{{ version }}/pandoc-{{ version }}-2-amd64.deb
```

Note the incremented build number. So future PRs will likely have to decrease this back to `1`. Should we template this to make it clear:

```
{% set deb_build_version = "1" %}
```

Perhaps this is overkill since the sha256 sums always need updating and are in the same section.

***

NOTE to maintainer: consider squash merging this PR so we don't have multiple commits.

**Comment by dhimmel** [comment]
Will have to rebase this PR after https://github.com/conda-forge/pandoc-feedstock/pull/20 is merged which performed a `conda smithy rerender` to fix the failing builds here.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,11 +29,12 @@ test:
...
-  home: http://johnmacfarlane.net/pandoc/
+  home: http://pandoc.org/
   license: GPL-2.0
   summary: 'Universal markup converter (repackaged binaries).'
 
 extra:
   recipe-maintainers:
     - janschulz
     - ocefpaf
+    - dhimmel
```
Comment (janschulz) [inline recipe/meta.yaml]: +1

Comment (ocefpaf) [reply recipe/meta.yaml]: +2

**Comment by ocefpaf** [comment]
> Will have to rebase this PR after https://github.com/conda-forge/pandoc-feedstock/pull/20 is merged which performed a `conda smithy rerender` to fix the failing builds here.

@dhimmel sorry for not reading this before. Next time you can use https://conda-forge.org/docs/webservice.html instead of opening a new PR.

**Comment by dhimmel** [comment]
Pandoc [released](https://github.com/jgm/pandoc/releases/tag/2.0.0.1) `v2.0.0.1` with some additional bug fixes compared to `v2.0`. Will it be problematic that the levels of semantic versioning has changed? Should I switch the PR to `v2.0.0.1`?

**Comment by ocefpaf** [comment]
> Pandoc [released](https://github.com/jgm/pandoc/releases/tag/2.0.0.1) `v2.0.0.1` with some additional bug fixes compared to `v2.0`. Will it be problematic that the number of semantic versioning levels has changed?

Not really.

> Should I switch the PR to `v2.0.0.1`?

Yes please :grimacing:

**Comment by dhimmel** [comment]
CI builds are passing for `v2.0.0.1`

**Pull request merged**